### PR TITLE
Improve python handler resolving

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
Make the resolving of python handlers more accurate to match Lambda

## Related Issue(s)
#372 

## Testing
New tests

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt [hunk 1]
```diff
@@ -3,46 +3,121 @@
...
+
+        val parentFolders = if (isReferencedByModule) {
+            remainder.substringBeforeLast("/", "").nullize(true)?.split('/') ?: emptyList()
+        } else {
+            remainder.split('/')
+        }
+
+        // Find the module by the name
+        val modules = PyModuleNameIndex.find(moduleName, project, true)
+        for (pyModule in modules) {
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt]: Nit: `modules.foreach` (unless the `continue` / `return` makes that unwieldy)

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt [hunk 2]
```diff
@@ -3,46 +3,121 @@
...
+
+    private fun validateHandlerPath(
+        module: Module?,
+        pyModule: PyFile,
+        isReferencedByModule: Boolean,
+        parentModules: List<String>,
+        parentFolders: List<String>
+    ): Boolean {
+        // Start matching to see if the parent folders align
+        var directory = pyModule.containingDirectory
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt]: So many `var`s and old school for loops - it hurts my eyes. ;)

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt]: I like the old school loop more than `return@reallyLongName` :-/

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/python/PythonLambdaHandlerResolver.kt]: Switched back to `.foreach`, originally the helper method was embedded in it so it had nested loops. Now that it is 1 layer deep, `.foreach` is clean again

**Pull request merged**